### DISCUSSION TITLE
Dynamic golden layout updateMethod

### DISCUSSION THREAD

**Discussion by ghost** [evennia/evennia#2484] [Evennia Questions] [answered]
Hello all, first time poster here.

So the multimedia plugin update was amazing. The ability to post images to specific golden layout windows got me inspired to do some weird shit.

But for my idea to work, I want to be able to dynamically change the state of a window's updateMethod on a msg-by-msg basis. So that, ultimately I could do something like this:
```
# Send two messages to a window, updating by 'newline', then clear the messages by sending one with 'replace'
target.msg(("Sample text.", {'type':'window1', 'updateMethod':'newline'} ))
target.msg(("With an addition.", {'type':'window1', 'updateMethod':'newline'} ))
target.msg(("And now a replacement.", {'type':'window1', 'updateMethod':'replace'} ))
```

Where should I start with this? How does Evennia's `.msg` share its 'type' to the goldenlayout.js plugin's componentState 'types'? Is there an easier/lazy way to do this? Thoughts, advice, or ideas are all much appreciated.

**Comment by friarzen** [comment, marked as answer]
Hmm...this isn't something the server currently has any control over.  `updateMethod` handling is all client-side manipulation done in the goldenlayout plugin, so this will need a bit of modification to that plugin to make it work.  Basically, you should hunt through and modify the `onText` ->  `addMessageToPaneDiv` function to add the `updateMethod` kwarg handling there.  You'd just need to check if the kwarg was provided and use that instead of the default behavior as a special case check.

**Reply by ghost** [reply]
This puts me in the right direction, thanks!
